### ISSUE TITLE
Clarification - Where can I see validation loss and training loss

### ISSUE THREAD

**Issue by kmsravindra** [AlexeyAB/darknet#983]
@AlexeyAB, Could you please clarify 
1. If I can see both the validation loss and the training loss metrics in the console while training? If so, can you please tell me which of those is where to look for in the console output?
2. On the chart, I see only the blue curve. I am not seeing other curve. Could you tell me if there is a provision to see both valid and train losses in the chart graph?

Thanks!

**Comment by AlexeyAB** [comment]
1. `valid=` in the obj.data is for `./darknet detector map`... - this indicator is much important than valid-loss

2. There is no simply way to do this - you should load validate images/labels and use forward_backward() without update(). But much better to get mAP on validation dataset for each 1000 iterations, may be I will add it.

**Comment by kmsravindra** [comment]
@AlexeyAB, Thank you. So is the 'map' calculated on the validation data mentioned in valid path?

**Comment by kmsravindra** [comment]
Thanks for the pointer. Understood.

**Comment by kmsravindra** [comment]
@AlexeyAB , When I ran the detector map, I got the below metrics - 
detections_count = 1947, unique_truth_count = 633  
 for thresh = 0.25, precision = 0.90, recall = 0.79, F1-score = 0.84 
 for thresh = 0.25, TP = 497, FP = 54, FN = 136, average IoU = 71.37 % 
 mean average precision (mAP) = 0.804040, or 80.40 % 

From here, 
I understood unique_truth_count = Total number of ground truth objects present in the actual data. 
But could you tell me what exactly is detections_count value referring to? Is it the value before NMS or after applying NMS and if so, then at what threshold was considered for the NMS and where is that cut-off mentioned?

**Comment by AlexeyAB** [comment]
1. `detections_count = 1947` is for `-thresh 0.005` - it is only for building PR-Curve and calculate mAP and AP: https://github.com/AlexeyAB/darknet/blob/17520296c730c7d7e2683452b11bf50fc8959688/src/detector.c#L586

2. `ignore_thresh=0.7` is only for training - it cancels negative delta for `objectness` if detection has `IoU(box,closest_truth)>ignore_thresh` : https://github.com/AlexeyAB/darknet/blob/17520296c730c7d7e2683452b11bf50fc8959688/src/yolo_layer.c#L219-L221

**Comment by kmsravindra** [comment]
Thanks for the clarification. I assume then this thresh 0.005 is the class confidence score.

**Comment by arslanan** [comment]
Hi, 

I know this subject is a bit old but I wasn't able to find another one dealing with this matter.
Is there now a way to plot both curve ?

If not, I would like to know if there is litterature explaining why it is "much better to get mAP on validation dataset for each 1000 iterations" than plotting the curve.

Thanks a lot !
